pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -128,6 +132,24 @@ func (r *NSTemplateSetReconciler) addFinalizer(nsTmplSet *toolchainv1alpha1.NSTe
 	return nil
 }
 
+// addLabels sets the labels for NSTemplateSet
+func (r *NSTemplateSetReconciler) addLabels(nsTmplSet *toolchainv1alpha1.NSTemplateSet) error {
```

Comment:
We don't need the provider key in the NSTemplateSet resources itself. We only need to add this label to all resources created by the NSTemplateSet.